codepipeline.tf

```diff
@@ -0,0 +1,117 @@
+// IAM
+resource "aws_iam_role" "default" {
+  name = "CodePipelineRole"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect    = "Allow"
+        Principal = { Service = "codepipeline.amazonaws.com" }
+        Action    = "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "default" {
+  role = aws_iam_role.default.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = "s3:*"
+        Resource = [
+          aws_s3_bucket.appspec_artifacts[0].arn,
+          "${aws_s3_bucket.appspec_artifacts[0].arn}/*"
+        ]
+      },
+      {
+        Effect   = "Allow"
+        Action   = "codedeploy:*"
+        Resource = "*"
+      },
+      {
+        Effect   = "Allow"
+        Action   = "codepipeline:*"
+        Resource = "*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "ecs:RegisterTaskDefinition",
+          "ecs:DescribeTaskDefinition",
+          "ecs:UpdateService",
+          "ecs:DescribeServices"
+        ]
+        Resource = "*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "elasticloadbalancing:DescribeTargetGroups",
+          "elasticloadbalancing:DescribeListeners",
+          "elasticloadbalancing:ModifyListener",
+          "elasticloadbalancing:DescribeRules",
+          "elasticloadbalancing:ModifyRule"
+        ]
+        Resource = "*"
+      },
+      {
+        Effect   = "Allow"
+        Action   = ["iam:PassRole"]
+        Resource = aws_iam_role.ecs_exec[0].arn
+      }
+    ]
+  })
+}
+
+// CodePipeline
+resource "aws_codepipeline" "default" {
```

Comment:
Resources should be conditional on whether the service actually needs a code pipeline. We don't want to be managing this stuff for every single service.